integration_tests/src/main/python/parquet_write_test.py

```diff
@@ -544,6 +544,19 @@ def test_fallback_to_single_writer_from_concurrent_writer(spark_tmp_path, aqe_en
             {"spark.sql.adaptive.enabled": aqe_enabled},
         ))
 
+def test_write_exceed_max_file_counter_error(spark_tmp_path):
+    gen_list = [("c1", BooleanGen())]
+    data_path = spark_tmp_path + 'PARQUET_DATA'
+    assert_gpu_error(
```

Comment:
I don't check the CPU version here. Because the logic in Spark is very time-consuming. It increments `file_counter` and check it each time until beyond the `MAX_FILE_COUNTER`: 
https://github.com/apache/spark/blob/3a3bc77f3dea368ca0b434a3f8a9629b5d69a5ca/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatDataWriter.scala#L166-L179

Reply:
We may need more discussion about how to test this feature for CPU.